src/go/k8s/pkg/resources/secret.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package resources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	redpandav1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	"github.com/redpanda-data/redpanda/src/go/k8s/pkg/labels"
+)
+
+const (
+	postStartKey = "postStart.sh"
+	preStopKey   = "preStop.sh"
+)
+
+var _ Resource = &SecretResource{}
+
+// SecretResource contains definition and reconciliation logic for operator's Secret
```

Comment:
just curious: why secret and not configmap?

Reply:
nit: this is not generic secret resource so I would probably name it a PreStartStopScriptResource or sth along those lines

or decouple the scripts and make it generic secretresource and inject the value via constructor

Reply:
changed